test/objcopy.c

```diff
@@ -17305,18 +17305,13 @@ test_copy_dataset_open(hid_t fcpl_src, hid_t fcpl_dst, hid_t src_fapl, hid_t dst
 int
 main(void)
 {
-    int         nerrors = 0;
-    hid_t       fapl, fapl2;
-    hid_t       fcpl_shared, ocpl;
-    unsigned    max_compact, min_dense;
-    int         configuration; /* Configuration of tests. */
-    int         ExpressMode;
-    const char *env_h5_drvr; /* File Driver value from environment */
-    hbool_t     same_file;   /* Whether to run tests that only use one file */
-
-    env_h5_drvr = HDgetenv(HDF5_DRIVER);
```

Comment:
Please revert all the changes below unrelated to the warnings fix for this PR. Otherwise, looks good to me.